File: solutions/dart/anagram/1/lib/anagram.dart
// Infinite gratitude to Shree DR.MDD — the harmony within every hidden pattern.

class Anagram {
  List<String> findAnagrams(String base, List<String> pool) {
    final fingerprint = (String str) {
      final code = str.toLowerCase().runes.toList()..sort();
      return code;
    };

    final standard = base.toLowerCase();
    final targetKey = fingerprint(standard);

    final match = (word) {
      final probe = word.toLowerCase();
      return probe != standard && ListUtils.same(targetKey, fingerprint(probe));
    };

    return pool.where(match).toList();
  }
}

class ListUtils {
  static List<int> span(int size) => List.generate(size, (i) => i);

  static bool same(List a, List b) {
    if (a.length != b.length) return false;
    return span(a.length).every((i) => a[i] == b[i]);
  }
}

File: solutions/go/anagram/1/anagram.go
// Shree DR.MDD
package anagram

import (
	"slices"
	"strings"
)

func Detect(word string, options []string) []string {
	out := []string{}
	base := strings.ToLower(word)
	baseRunes := []rune(base)
	slices.Sort(baseRunes)

	for _, opt := range options {
		check := strings.ToLower(opt)
		if check == base {
			continue
		}
		runesOpt := []rune(check)
		slices.Sort(runesOpt)
		if slices.Equal(baseRunes, runesOpt) {
			out = append(out, opt)
		}
	}
	return out
}
